test/extended/networking/commatrix.go

```diff
@@ -0,0 +1,252 @@
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const (
+	minimalDocCommatrixVersion = 4.16
+	docCommatrixBaseUrl        = "https://raw.githubusercontent.com/openshift/openshift-docs/enterprise-VERSION/snippets/network-flow-matrix.csv"
+	diffFileComments           = "// `+` indicates a port that isn't in the current documented matrix, and has to be added.\n" +
+		"// `-` indicates a port that has to be removed from the documented matrix.\n"
+)
+
+var (
+	// Entries which are open on the worker node instead of master in standard cluster.
+	// Will be excluded in diff generatation between documented and generated comMatrix.
+	StandardExcludedMasterComDetails = []types.ComDetails{
+		{
+			Direction: "Ingress",
+			Protocol:  "TCP",
+			Port:      80,
+			NodeRole:  "master",
+			Service:   "router-default",
+			Namespace: "openshift-ingress",
+			Pod:       "router-default",
+			Container: "router",
+			Optional:  false,
+		}, {
+			Direction: "Ingress",
+			Protocol:  "TCP",
+			Port:      443,
+			NodeRole:  "master",
+			Service:   "router-default",
+			Namespace: "openshift-ingress",
+			Pod:       "router-default",
+			Container: "router",
+			Optional:  false,
+		},
+	}
+)
+
+var _ = g.Describe("[sig-network][Feature:commatrix][Serial]", func() {
+	g.It("generated communication matrix should be equal to documented communication matrix", func() {
+		artifactsDir := filepath.Join(exutil.ArtifactDirPath(), "commatrix")
+
+		err := os.MkdirAll(artifactsDir, 0755)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		cs, err := client.New()
+		o.Expect(err).ToNot(o.HaveOccurred())
+
+		configClient := configv1client.NewForConfigOrDie(cs.Config)
+
+		deployment := types.Standard
+		isSNO, err := isSNOCluster(configClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isSNO {
+			deployment = types.SNO
+		}
+
+		env := types.Cloud
+		isBM, err := isBMCluster(configClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		if isBM {
+			env = types.Baremetal
+		}
+
+		g.By("get cluster's version and check if it's suitable for test")
+		clusterXYVersion, err := getClusterXYersion(configClient)
+		o.Expect(err).NotTo(o.HaveOccurred())
+		floatClusterXYVersion, err := strconv.ParseFloat(clusterXYVersion, 64)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		if floatClusterXYVersion < minimalDocCommatrixVersion {
+			g.Skip(fmt.Sprintf("If the cluster version is lower than the lowest version that "+
+				"has a documented communication matrix (%v), skip test", minimalDocCommatrixVersion))
+		}
+		docCommatrixVersionedUrl := strings.Replace(docCommatrixBaseUrl, "VERSION", clusterXYVersion, 1)
+		resp, err := http.Get(docCommatrixVersionedUrl)
+		o.Expect(err).ToNot(o.HaveOccurred())
+		defer resp.Body.Close()
+		// expect response status code to differ from status not found
+		docURLStatusIsNotFound := resp.StatusCode == http.StatusNotFound
+		o.Expect(docURLStatusIsNotFound).To(o.BeFalse())
```

Comment:
Expect(resp.StatusCode).To(Equal(http.StatusNotFound)))

Reply:
We want to differ `resp.StatusCode` from `http.StatusNotFound`.
Though `Expect(resp.StatusCode).ToNot(Equal(http.StatusNotFound)))` is a indeed better, thanks!